### DISCUSSION TITLE
How do you suggest we retrieve querystring or route values?

### DISCUSSION THREAD

**Discussion by simonmesmith** [reflex-dev/reflex#68] [Q&A] [answered]
Let's say I want to create something like a blog, where we have an identifier in the query string or route for a post.

I know how to do this in Flask, but I'm wondering if you have other thoughts on how to implement this with Pynecone.

My sense is that you've abstracted this the way you have with several other things, but I couldn't find any examples.

So, basically, let's say I have a list of items and want them to link to details. Or let's say I want people to be able to share URLs with dynamic querystring values. How do you suggest I implement this?

Thanks!

**Comment by FHU-yezi** [comment]
It seems that we can't do this now, but it should be consider in the comming versions. maybe we can add a variable in the root state iof the App? Just like `set_xxx`, do some magic things with `setattr()`.

**Comment by picklelo** [comment, marked as answer]
This isn't supported at the moment, but we have plans to add it soon. We have a ticket to track it [here](https://github.com/pynecone-io/pynecone/issues/63).

The design we had in mind currently that the render functions take in the query arguments. So if you had a blog page and you want the path to be `/blog/<post_id>` you could define your render function like:

```
def blog(post_id: pc.Var[str]):
  return pc.box(...)
```

Then you could add a route like
```
app.add_page(blog, path="/blog/:post_id")
```

For query parameters, we would make them accessible as kwargs.

```
def blog(post_id: pc.Var[str], **kwargs):
  ...
```

In the app state, we would provide some way to access the query parameters. So in the event handlers, maybe you could call `self.get_query_params()` and do some logic with it.

This design is still in process, I'll post updates on the linked ticket as we make progress. We're open to ideas here.

**Reply by FHU-yezi** [reply]
I think the query args may have mutli values for one key, such as `?foo=bar&foo=bar1`, I'm new to frond-end developing, is this true? How we can deal with this? Just ignore the dumplicate args is fine, or we can make a object to the query args, and we use `query_args[key]` to get the first value of the key, use `query_args.get(key)` do just the same thing, similar to the dict in Python. If we want to get access to other values, use `query_args.get_list(key)`, and we got a `List[Any]`.

**Reply by picklelo** [reply]
Generally there should only be one value per key, I believe. We plan to use the same approach [NextJS uses](https://nextjs.org/docs/routing/dynamic-routes). If there are more complex use cases people would like, we can look into that too.

**Reply by matt-grain** [reply]
definitively a must have!
